File: frontend/src/lib/models/mealPlanEntry.ts
/**
 * Represents a single recipe planned for a specific date on the frontend.
 * This should mirror the backend's `models.MealPlanEntry` struct.
 */
export interface MealPlanEntry {
	id: string;         // Unique ID for this meal plan entry
	date: string;       // The specific date as an ISO string (e.g., "2023-10-26T00:00:00Z")
	recipe_id: string;  // ID of the planned recipe
	created_at: string; // Timestamp of when the entry was created (ISO string)
	// recipe?: Recipe; // Optional: To hold hydrated recipe details if fetched
}

// You might also want to include the Recipe type here or import it if it's used often with MealPlanEntry
// For now, keeping it separate as per current store logic.
// import type { Recipe } from '$lib/types';

File: backend/internal/models/mealplanentry.go
package models

import "time"

// MealPlanEntry represents a single recipe planned for a specific date.
// Each assignment of a recipe to a day is a unique entry.
type MealPlanEntry struct {
	ID        string    `json:"id"`         // Unique ID for this meal plan entry (e.g., UUID)
	Date      time.Time `json:"date"`       // The specific date (YYYY-MM-DD), time part normalized to UTC midnight
	RecipeID  string    `json:"recipe_id"`  // ID of the planned recipe
	CreatedAt time.Time `json:"created_at"` // Timestamp of when the entry was created
	// UserID    string    `json:"user_id"`    // Future: For multi-user support
}
